viadot/tasks/azure_data_lake.py

```diff
@@ -567,10 +572,24 @@ def run(
 
         full_dl_path = os.path.join(credentials["ACCOUNT_NAME"], path)
 
-        self.logger.info(f"Listing files in {full_dl_path}...")
-        files = lake.ls(path)
-        self.logger.info(f"Successfully listed files in {full_dl_path}.")
+        self.logger.info(f"Listing files in {full_dl_path}.")
+        if recursive:
+            self.logger.info("Loading ADLS directories recursively.")
+            files = lake.find(path)
+            if file_to_match:
+                conditions = [file_to_match in item for item in files]
+                valid_files = np.array([])
+                if any(conditions):
+                    index = np.where(conditions)[0]
+                    files = list(np.append(valid_files, [files[i] for i in index]))
+                else:
+                    raise FileExistsError(
```

Comment:
Wrong error class, as peer definition: "Raised when trying to create a file or directory which already exists. Corresponds to errno [EEXIST](https://docs.python.org/3/library/errno.html#errno.EEXIST)." (https://docs.python.org/3/library/exceptions.html#FileExistsError), sounds exactly the opposite of what's happening here.